libmozevent/utils.py

```diff
@@ -18,15 +19,22 @@
 
 def run_tasks(awaitables: Iterable):
     """
-    Helper to run tasks concurrently, but when an exception is raised
-    by one of the tasks, the whole stack stops.
+    Helper to run tasks concurrently
+    When an exception is raised by one of the tasks or a SIGTERM signal is received,
+    the whole stack stops and an asyncio.CancelledError exception is raised
     """
+    event_loop = asyncio.get_event_loop()
+
+    # Create a task grouping all awaitables and running them concurrently
+    task = asyncio.gather(*awaitables)
```

Comment:
You move the `gather` call out of the try/except. I'm not sure it was super important before, as most exceptions would happen on the `await`, but it would be better to preserve the previous setup.

Reply:
`asyncio.gather` returns an awaitable, it should never fail until something that is not an awaitable is passed (which should of course not happen).
I have moved some code to cancel the main task, I'll test to ensure the behavior is the same.

Reply:
Hmm, ok i broke some things

Reply:
I think the initial patch was cleaner, I'll try to add another try/catch block at the start of the function